client/src/app/site/motions/services/motion-import.service.ts

```diff
@@ -0,0 +1,435 @@
+        'origin', 'motion block'];
+
+    public columnSeparator = '';
+    public encodings = ['utf-8', 'iso-8859-1', 'iso-8859-15'];
+    public columnSeparators : {name: string, value: string}[] = [
+        {name: 'Comma', value: ','},
+        {name: 'Semicolon', value: ';'},
+        // {name: 'Tabulator', value: '\t'},
+        {name: 'Automatic', value: '' }
+    ];
+
+    private reader: FileReader;
+
+    public newEntries = new BehaviorSubject<NewMotionEntry[]>([]);
+    private _entries: NewMotionEntry[] = [];
+
+    public errorEvent = new EventEmitter<string>();
+
+    private _preview: ImportMotionCSVPreview;
+
+    public get summary(): ImportMotionCSVPreview {
+        if (!this._preview) {
+            this.updatePreview();
+        }
+        return this._preview;
+    }
+
+    /**
+     * Constructor. Creates a fileReader to subscribe to it for incoming parsed
+     * strings
+     * @param categoryRepo Repository to fetch pre-existing categories
+     * @param motionBlockRepo Repository to fetch pre-existing motionBlocks
+     * @param userRepo Repository to query/ create users
+     * @param translate Translation service
+     * @param papa External csv parser (ngx-papaparser)
+     */
+    public constructor( private repo: MotionRepositoryService,
+        private categoryRepo: CategoryRepositoryService,
+        private motionBlockRepo: MotionBlockRepositoryService,
+        private userRepo: UserRepositoryService,
+        private translate: TranslateService,
+        private papa: Papa
+    ) {
+        this.reader = new FileReader();
+        const me = this;
+        this.reader.onload = function(event: any): void {
+            me.parseInput(event.target.result);
+        };
+    }
+
+    /**
+     * Parses the data input. Expects a string as returned by via a
+     * File.readAsText() operation
+     * TODO:
+     * @param file
+     */
+    public parseInput(file: string): void {
+        this._entries = [];
+        const papaConfig : PapaParseConfig = {
+            header: false,
+            skipEmptyLines: true
+        };
+        if (this.columnSeparator) {
+            papaConfig.delimiter = this.columnSeparator;
+        }
+        const entry_lines = this.papa.parse(file, papaConfig).data;
+        if ( !this.checkHeader(entry_lines.shift()) ) {
+            this.errorEvent.emit('File could not be parsed');
+            this.clearPreview();
+        } else {
+            entry_lines.forEach( line => {
+
+                const model : Partial<CreateMotion> = {};
+                let submitters: { user_id: number[], user_names: string[], newUsers: string[] } ;
+                let category: { category_id?: number, category_name?: string, newCategory?: string };
+                let motion_block: { block_id?: number, block_name?: string, newBlock?: string };
+
+                for (let idx = 0; idx < this.expectedHeader.length; idx++ ) {
```

Comment:
Note: You need to have the limit of the loop to be the minimum of expectedHeaders and "real" headers.